### ISSUE TITLE
SCM bad replacement for non Conanfiles

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#5777]
```python

class Something(object):
     scm = {"url": "auto"}


class Pkg(Something, Conanfile):
     pass
```

This does a replacement in the Something class, instead of only in the Pkg class, which is a bug. If a good replacement could be done, that would be a more elegant way for extending classes using python_requires.

It is NOT possible right now to implement this with the current ast replacement. I'd suggest to dump the resulting SCM information to conandata.yml file, and recover it from there, not modifying the conanfile anymore.

**Comment by memsharded** [comment]
@SSE4 please contribute a PR to "develop2" branch that:

- Removes "scm_to_conandata" from config.
- Removes the modification of recipe, and always dumps the scm data to conandata.yml

Do another PR to "develop", that raises ``conan_v2_error()`` if ``scm_to_conandata`` is not enabled.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/8446 removes this behavior, conanfile.py will never be modified.
This will be released in Conan 2.0
